shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -244,6 +244,16 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
   [self updateEditingState];
 }
 
+- (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
+  if (self.returnKeyType == UIReturnKeyDone && [text isEqualToString:@"\n"]) {
```

Comment:
This will trigger on paste of a string containing exactly one newline char, but I can't see a nice way in UIKit APIs to intercept a `Done` pressed event directly. If you're aware of one, I'd be super happy to hear about it :)